src/validation.cpp

```diff
@@ -4296,42 +4298,49 @@ void CChainState::LoadExternalBlockFile(
                     continue;
             } catch (const std::exception&) {
                 // no valid block header found; don't complain
+                // (this happens at the end of every blk.dat file)
                 break;
             }
             try {
-                // read block
-                uint64_t nBlockPos = blkdat.GetPos();
+                // read block header
+                const uint64_t nBlockPos = blkdat.GetPos();
```

Comment:
nit: could use list initialization?

```suggestion
                const uint64_t nBlockPos{blkdat.GetPos()};
```